### ISSUE TITLE
Feature discussion: replay

### ISSUE THREAD

**Issue by rixx** [venueless/venueless#131]
So currently, speakers can't really follow the chat and emoji reactions to their talks, and while they can scroll back the chat, this is not giving them the appropriate context. Twitch-like chat-synced replay would be amazing.

Now, different from vue, we currently don't have a platform where recordings hang out to stay, nor are we planning to have one. Instead I propose a data export plus a display wrapper. The data export contains at least chat, ideally also emoji reactions, over a certain range. The display wrapper script syncs those up with the playback of a recording, to be embedded on event pages or whereever.

I know that this feature has a rather large maintenance burden (new chat features need to be included there) for relatively little on-platform value, but it would make the platform as a whole less ephemeral and more attractive to use.

**Comment by raphaelm** [comment]
Do you have ideas how the sync could work? We'd need some way to know which point in (venueless server time) is the initial cut of the recording (which usually will be later than the start of stream input to the platform).

**Comment by rixx** [comment]
Yeah, I got nothing. Ideally the stream handler could tell us the timestamp. Adding a way to do a final offset-adjustment to the component wouldn't be too bad, so you could do it manually off an export of reasonable range.

**Comment by raphaelm** [comment]
sorry, mausgerutscht

**Comment by rashfael** [comment]
Might be difficult if we have no control over the recording or cdn. Also not sure if time drifting in long hls content is a thing.

**Comment by rixx** [comment]
We don't need control over the recording. My suggested workflow is:

- Stream (external or whatever) runs, with chat
- After the event, the stream handler supplies the organiser with a recording and names a starting time and duration
- We supply a data export and script for the site hosting the video
- Final adjustments by ~seconds can be made by the script
